nixos/modules/services/security/keylime.nix

```diff
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.keylime;
+in
+{
+  options = {
+    services.keylime = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = lib.mdDoc "Whether to enable keylime services.";
+      };
+
+      registrar = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = lib.mdDoc "Whether to enable keylime-registrar service.";
+        };
+      };
+
+      verifier = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = lib.mdDoc "Whether to enable keylime-verifier service.";
+        };
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "keylime";
+        description = lib.mdDoc "User which runs keylime.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "keylime";
+        description = lib.mdDoc "Group which runs keylime.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      environment.etc."keylime/tenant.conf".source = "${pkgs.keylime}/etc/config/tenant.conf";
+
+      users.users.keylime = {
+        description = "Keylime user";
+        extraGroups = [ "${cfg.group}" ];
+        isSystemUser = true;
+      };
+
+      users.groups.${cfg.group} = {};
```

Comment:
While I don't think there's any documentation saying this, I believe it's common not to create groups dynamically like this.

While you could make an argument that it makes it more fault tolerant in case the user forgot to declare the group, I'd rather the config throw an error at me than letting it silently go through. I think most of the cases where the group doesn't exist, it's either due to an error or a typo.